scripts/convert_to_tub_v2.py

```diff
@@ -22,29 +22,44 @@
 
 
 def convert_to_tub_v2(paths, output_path):
+    if type(paths) is str:
+        paths = [paths]
     legacy_tubs = [LegacyTub(path) for path in paths]
     output_tub = None
-    print('Total number of tubs: %s' % (len(legacy_tubs)))
+    print(f'Total number of tubs: {len(legacy_tubs)}')
 
     for legacy_tub in legacy_tubs:
         if not output_tub:
-            output_tub = Tub(output_path, legacy_tub.inputs, legacy_tub.types, list(legacy_tub.meta.items()))
+            output_tub = Tub(output_path, legacy_tub.inputs,
+                             legacy_tub.types, list(legacy_tub.meta.items()))
 
         record_paths = legacy_tub.gather_records()
         bar = IncrementalBar('Converting', max=len(record_paths))
-
+        prev_rec_num = None
         for record_path in record_paths:
             try:
                 contents = Path(record_path).read_text()
                 record = json.loads(contents)
                 image_path = record['cam/image_array']
+                rec_num = int(image_path.split('_')[0])
                 image_path = os.path.join(legacy_tub.path, image_path)
                 image_data = Image.open(image_path)
                 record['cam/image_array'] = image_data
-                output_tub.write_record(record)
+                # first record or they are continuous, just append
+                if not prev_rec_num or rec_num == prev_rec_num + 1:
+                    output_tub.write_record(record)
+                    prev_rec_num = rec_num
+                # otherwise fill the gap with dummy records
+                else:
+                    prev_rec_num += 1
+                    while prev_rec_num < rec_num:
+                        idx = output_tub.manifest.current_index
+                        output_tub.write_record({})
```

Comment:
Define a `dict` `EMPTY`  which looks something like:

```python
{'__empty__': true}
```
Write that for every deleted record.  That way the intention here is very clear when reading though the file.

Reply:
Good point. And I think this logic should be part of the tub itself, not the user of the function. So we rather define a specific empty record in the tub, that will be generated in the same consistent way by whoever needs to add empty records.